# Phase 2: Dataset Understanding
## Objective

The objective of this notebook is to understand the structure, features, and characteristics of the employee dataset before performing any data quality assessment or analysis.

In [2]:
import pandas as pd
df=pd.read_csv("../data/raw/WA_Fn-UseC_-HR-Employee-Attrition.csv")
df.shape

(1470, 35)

The dataset contains 1,470 employee records and 35 features, providing sufficient information for workforce analytics.

In [3]:
df.head(10)

,Age,Attrition,BusinessTravel,DailyRate,Department,DistanceFromHome,Education,EducationField,EmployeeCount,EmployeeNumber,...,RelationshipSatisfaction,StandardHours,StockOptionLevel,TotalWorkingYears,TrainingTimesLastYear,WorkLifeBalance,YearsAtCompany,YearsInCurrentRole,YearsSinceLastPromotion,YearsWithCurrManager
0,41,Yes,Travel_Rarely,1102,Sales,1,2,Life Sciences,1,1,...,1,80,0,8,0,1,6,4,0,5
1,49,No,Travel_Frequently,279,Research & Development,8,1,Life Sciences,1,2,...,4,80,1,10,3,3,10,7,1,7
2,37,Yes,Travel_Rarely,1373,Research & Development,2,2,Other,1,4,...,2,80,0,7,3,3,0,0,0,0
3,33,No,Travel_Frequently,1392,Research & Development,3,4,Life Sciences,1,5,...,3,80,0,8,3,3,8,7,3,0
4,27,No,Travel_Rarely,591,Research & Development,2,1,Medical,1,7,...,4,80,1,6,3,3,2,2,2,2
5,32,No,Travel_Frequently,1005,Research & Development,2,2,Life Sciences,1,8,...,3,80,0,8,2,2,7,7,3,6
6,59,No,Travel_Rarely,1324,Research & Development,3,3,Medical,1,10,...,1,80,3,12,3,2,1,0,0,0
7,30,No,Travel_Rarely,1358,Research & Development,24,1,Life Sciences,1,11,...,2,80,1,1,2,3,1,0,0,0
8,38,No,Travel_Frequently,216,Research & Development,23,3,Life Sciences,1,12,...,2,80,0,10,2,3,9,7,1,8
9,36,No,Travel_Rarely,1299,Research & Development,27,3,Medical,1,13,...,2,80,2,17,3,2,7,7,7,7


### Observation

The dataset contains employee demographic, job-related, compensation, and satisfaction information. Based on the first few records, the dataset appears to be structured; however, data quality checks will be performed in the next phase.

In [4]:
df.columns

Index(['Age', 'Attrition', 'BusinessTravel', 'DailyRate', 'Department',
       'DistanceFromHome', 'Education', 'EducationField', 'EmployeeCount',
       'EmployeeNumber', 'EnvironmentSatisfaction', 'Gender', 'HourlyRate',
       'JobInvolvement', 'JobLevel', 'JobRole', 'JobSatisfaction',
       'MaritalStatus', 'MonthlyIncome', 'MonthlyRate', 'NumCompaniesWorked',
       'Over18', 'OverTime', 'PercentSalaryHike', 'PerformanceRating',
       'RelationshipSatisfaction', 'StandardHours', 'StockOptionLevel',
       'TotalWorkingYears', 'TrainingTimesLastYear', 'WorkLifeBalance',
       'YearsAtCompany', 'YearsInCurrentRole', 'YearsSinceLastPromotion',
       'YearsWithCurrManager'],
      dtype='str')

### Employee Information

- Age
- Gender
- MaritalStatus

### Job Information

- Department
- JobRole
- JobLevel
- BusinessTravel

### Compensation

- MonthlyIncome
- DailyRate
- HourlyRate
- MonthlyRate

### Satisfaction Metrics

- JobSatisfaction
- EnvironmentSatisfaction
- RelationshipSatisfaction
- WorkLifeBalance

### Experience

- TotalWorkingYears
- YearsAtCompany
- YearsInCurrentRole
- YearsSinceLastPromotion

In [5]:
df.dtypes

Age                         int64
Attrition                     str
BusinessTravel                str
DailyRate                   int64
Department                    str
DistanceFromHome            int64
Education                   int64
EducationField                str
EmployeeCount               int64
EmployeeNumber              int64
EnvironmentSatisfaction     int64
Gender                        str
HourlyRate                  int64
JobInvolvement              int64
JobLevel                    int64
JobRole                       str
JobSatisfaction             int64
MaritalStatus                 str
MonthlyIncome               int64
MonthlyRate                 int64
NumCompaniesWorked          int64
Over18                        str
OverTime                      str
PercentSalaryHike           int64
PerformanceRating           int64
RelationshipSatisfaction    int64
StandardHours               int64
StockOptionLevel            int64
TotalWorkingYears           int64
TrainingTimesL

The dataset contains both numerical and categorical features, making it suitable for statistical analysis as well as segmentation-based analysis.

## Target Variable

Target Variable:
Attrition

Description:
Indicates whether an employee has left the organization.

Possible Values:
• Yes
• No

In [6]:
df.isnull().sum()

Age                         0
Attrition                   0
BusinessTravel              0
DailyRate                   0
Department                  0
DistanceFromHome            0
Education                   0
EducationField              0
EmployeeCount               0
EmployeeNumber              0
EnvironmentSatisfaction     0
Gender                      0
HourlyRate                  0
JobInvolvement              0
JobLevel                    0
JobRole                     0
JobSatisfaction             0
MaritalStatus               0
MonthlyIncome               0
MonthlyRate                 0
NumCompaniesWorked          0
Over18                      0
OverTime                    0
PercentSalaryHike           0
PerformanceRating           0
RelationshipSatisfaction    0
StandardHours               0
StockOptionLevel            0
TotalWorkingYears           0
TrainingTimesLastYear       0
WorkLifeBalance             0
YearsAtCompany              0
YearsInCurrentRole          0
YearsSince

In [7]:
df.duplicated().sum()

np.int64(0)

## Feature Classification

Identifier

- EmployeeNumber

Categorical Features

- Gender
- Department
- JobRole
- BusinessTravel
- MaritalStatus

Numerical Features

- Age
- MonthlyIncome
- DailyRate
- TotalWorkingYears
- YearsAtCompany

Target Variable

- Attrition

## Business Relevance

The dataset provides employee demographic, compensation, experience, and satisfaction information, enabling the HR team to identify patterns associated with employee attrition and workforce retention.


### Observation
- The Column can be categorized into groups which can help us understand the cause of attration in more detail
- Datatypes shows us the DataType of every column or the datatype each column uses
- There are 0 missing values which tell us that the Data is very intact
- There aren't any duplicate values in the DataSet

## Analyst Notes

### Key Findings

- Dataset contains 1,470 employee records.
- Attrition is the target variable.
- Both numerical and categorical variables are available.
- Dataset structure is appropriate for workforce analytics.

### Risks

- Data quality has not yet been assessed.

### Next Phase

Proceed to Data Profiling to evaluate completeness, consistency, and quality.